packages/opentelemetry/src/contextManager.ts

```diff
@@ -1,14 +1,28 @@
 import type { Context, ContextManager } from '@opentelemetry/api';
-import type { Carrier, Hub } from '@sentry/core';
-import { getCurrentHub, getHubFromCarrier } from '@sentry/core';
-import { ensureHubOnCarrier } from '@sentry/core';
+import type { Hub } from '@sentry/core';
+import { getCurrentHub } from '@sentry/core';
+import { SENTRY_FORK_ISOLATION_SCOPE_CONTEXT_KEY } from './constants';
+import { OpenTelemetryHub } from './custom/hub';
 
 import { setHubOnContext } from './utils/contextData';
 
-function createNewHub(parent: Hub | undefined): Hub {
-  const carrier: Carrier = {};
-  ensureHubOnCarrier(carrier, parent);
-  return getHubFromCarrier(carrier);
+function createNewHub(parent: Hub | undefined, shouldForkIsolationScope: boolean): Hub {
```

Comment:
`ensureHubOnCarrier` used to _always_ fork the isolation scope, which is not what we want here. Now, we only fork this if `withIsolationScope` is used.